addons/binding/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/eep/A5_20/A5_20_04.java

```diff
@@ -0,0 +1,272 @@
+                }
+            }
+        }
+
+        return (byte) ((value - 10.0) * (255.0 / 20.0));
+    }
+
+    private byte getMc(Map<String, State> currentState) {
+        State current = currentState.get(CHANNEL_MEASUREMENT_CONTROL);
+
+        if ((current != null) && (current instanceof OnOffType)) {
+            OnOffType state = current.as(OnOffType.class);
+
+            if (state != null) {
+                return (byte) (state.equals(OnOffType.ON) ? 0x00 : 0x40);
+            }
+        }
+
+        return 0x00; // on
+    }
+
+    private byte getWuc(Map<String, State> currentState) {
+        State current = currentState.get(CHANNEL_WAKEUPCYCLE);
+
+        if ((current != null) && (current instanceof DecimalType)) {
+            DecimalType state = current.as(DecimalType.class);
+
+            if (state != null) {
+                return (byte) (state.byteValue() & 0x3F);
+            }
+        }
+
+        return 0x13; // 19 = 600 sec = 10 min
+    }
+
+    private byte getDso(Map<String, State> currentState) {
+        State current = currentState.get(CHANNEL_DISPLAY_ORIENTATION);
+
+        if ((current != null) && (current instanceof DecimalType)) {
+            DecimalType state = current.as(DecimalType.class);
+
+            if (state != null) {
+                return (byte) (((state.byteValue() / 90) << 4) & 0x30);
+            }
+        }
+
+        return 0x00; // 0°
+    }
+
+    private byte getBlc(Map<String, State> currentState) {
+        State current = currentState.get(CHANNEL_BUTTON_LOCK);
+
+        if ((current != null) && (current instanceof OnOffType)) {
+            OnOffType state = current.as(OnOffType.class);
+
+            if (state != null) {
+                return (byte) (state.equals(OnOffType.ON) ? 0x04 : 0x00);
+            }
+        }
+
+        return 0x00; // unlocked
+    }
+
+    private byte getSer(Map<String, State> currentState) {
+        State current = currentState.get(CHANNEL_SERVICECOMMAND);
+
+        if ((current != null) && (current instanceof DecimalType)) {
+            DecimalType state = current.as(DecimalType.class);
+
+            if (state != null) {
+                return (byte) (state.byteValue() & 0x03);
+            }
+        }
+
+        return 0x00; // 0 = no change
+    }
+
+    @Override
+    protected void convertFromCommandImpl(String channelId, String channelTypeId, Command command,
+            Map<String, State> currentState, Configuration config) {
+
+        if (CHANNEL_SEND_COMMAND.equals(channelId) && (command.equals(OnOffType.ON))) {
+            byte db3 = getPos(currentState);
+            byte db2 = getTsp(currentState);
+            byte db1 = (byte) (0x00 | getMc(currentState) | getWuc(currentState));
+            byte db0 = (byte) (0x00 | getDso(currentState) | TeachInBit | getBlc(currentState) | getSer(currentState));
+
+            setData(db3, db2, db1, db0);
+
+            return;
+        }
+
+        if (command instanceof State) {

```

Comment:
Why do you check if command is a State? Item states should only be updated through the SensorHandler during a received messages.